uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PopupMediaItem.cpp

```diff
@@ -52,52 +61,49 @@ QString PopupMediaItem::title() const
 }
 
 /*!
-  \brief Returns the sourceUrl of the \c MediaPopupElement.
+  \brief Returns the caption of the \c MediaPopupElement.
  */
-QUrl PopupMediaItem::sourceUrl() const
+QString PopupMediaItem::caption() const
 {
-  return m_popupMedia->value()->sourceUrl();
+  return m_popupMedia->caption();
 }
 
 /*!
-  \brief Returns the linkUrl of the \c MediaPopupElement.
+  \brief Returns the popupMediaType of the \c MediaPopupElement.
  */
-QUrl PopupMediaItem::linkUrl() const
+PopupMediaType PopupMediaItem::popupMediaType() const
 {
-  return m_popupMedia->value()->linkUrl();
+  return m_popupMedia->popupMediaType();
 }
 
 /*!
-  \brief Returns the caption of the \c MediaPopupElement.
+  \brief Returns the PopupMedia of the \c MediaPopupElement.
  */
-QString PopupMediaItem::caption() const
+PopupMedia* PopupMediaItem::popupMediaItem() const
 {
-  return m_popupMedia->caption();
+  return m_popupMedia;
 }
 
-/*!
-  \brief Returns the popupMediaType of the \c MediaPopupElement.
- */
-PopupMediaType PopupMediaItem::popupMediaType() const
+QColor PopupMediaItem::jsonColorHelper(const QJsonArray& colorArray, const int index) const
 {
-  return m_popupMedia->popupMediaType();
+  if (!colorArray.isEmpty() && index < colorArray.size())
+  {
+    // color scheme is [Red, Green, Blue, Alpha]
+    // https://developers.arcgis.com/web-map-specification/objects/color
+    return QColor(colorArray[index][0].toInt(),
+                  colorArray[index][1].toInt(),
+                  colorArray[index][2].toInt(),
+                  colorArray[index][3].toInt());
```

Comment:
It's probably overkill, so this is optional, but consider something like this for readability.
```c++
const auto jsonColor = colorArray[index];
const auto r = colorArray[0].toInt();
const auto g = colorArray[1].toInt();
const auto b = colorArray[2].toInt();
const auto a = colorArray[3].toInt();
return QColor{r, g, b, a};
```

Might be too much for something as simple as this, so up to you. You could probably do some fancy looping, but that would be overkill IMO.

Reply:
Loop is definitely overkill but putting the values in descriptive variables is a good idea.